### PR TITLE
Stop changing 'encoding'

### LINKED ISSUE #144: Filename is wiped out when jumping to a buffer
Usually jumping to a buffer will immediately display the filename at the bottom of the screen. With editorconfig installed, the filename flashes and then disappears.

Commenting out Line 70 and 79 to prevent setting `&encoding`, fixes the issue but I'm wondering what consequences that has.
Using Vim 8.2 in Terminal/iTerm on MacOS 10.15

https://github.com/editorconfig/editorconfig-vim/blob/c5804257c6b38c5537b71ac91ed051c70eaeb5e5/autoload/editorconfig_core/ini.vim#L70-L79

### LINKED ISSUE #145: charset=utf-8 triggers double-encoding with fileencodings=latin1
Recently I added an .editorconfig with the following content to my ansible directory:
```ini
[*]
charset = utf-8
```

Usually I have the setting `set fileencodings=ucs-bom,latin1` in my vim config as most of my files are in latin1.

Now, when I open a yaml file inside the ansible tree that already has some utf-8 german umlauts in it, it gets opened as latin1 and then converted to utf-8 every time I open it. That leads to increasing lines with `Ã~CÂ~CÃ~BÂ¼` (in this example for `ü`)

I get the same content, when I (outside of that tree) open a file, put an german umlaut into it. Saving, leaving and reopening, then `set fenc=utf-8` `:w` and start over multiple times. So it seems to me that this addon just sets fenc when opening a file inside a tree. The correct way would be to set fileencodings prior to open the buffer or reopen it with `:e ++enc=utf8`

### PR DESCRIPTION
**Pull request by k-takata**
Changing 'encoding' has many side effects and that should be avoided.
Strip BOM manually.

Fix #144
Fix #145
Fix #147

### CODE REVIEW

File: autoload/editorconfig_core/ini.vim [hunk 1]
```diff
@@ -60,23 +60,24 @@ lockvar s:SECTCRE s:OPTCRE s:MAX_SECTION_NAME s:MAX_PROPERTY_NAME s:MAX_PROPERTY
...
         return {}
     endif
 
-    try     " so &encoding will always be reset
-        let &encoding = 'utf-8'     " so readfile() will strip BOM
+    try
         let l:lines = readfile(a:config_filename)
+        if &encoding !=? 'utf-8'
+            " strip BOM
+            if len(l:lines) > 0 && l:lines[0][:2] ==# "\xEF\xBB\xBF"
```
Comment (cxw42) [inline autoload/editorconfig_core/ini.vim:71]: Will the first three bytes always come through as the UTF-8 BOM values?  E.g., what happens when the file is UTF-16?

**Review by cxw42 — COMMENTED** [review]
Thanks for this PR!  Would you be willing to add some test cases of files with different encodings?  I can help with the test-case plumbing if that would be helpful.

File: autoload/editorconfig_core/ini.vim (hunk 1 above)
Comment (k-takata) [reply autoload/editorconfig_core/ini.vim:71]: This part strips the UTF-8 BOM only when it exists. Otherwise it is kept as is.

**Comment by k-takata** [comment]
> Would you be willing to add some test cases of files with different encodings?

Do we need to add some test cases with a `.editorconfig` file with UTF-8 BOM? Then, should https://github.com/editorconfig/editorconfig-plugin-tests be updated?

**Comment by cxw42** [comment]
> > Would you be willing to add some test cases of files with different encodings?
> 
> Do we need to add some test cases with a `.editorconfig` file with UTF-8 BOM? Then, should https://github.com/editorconfig/editorconfig-plugin-tests be updated?

@k-takata Good question!  I don't think we need to update the main plugin tests, since this problem is specific to this plugin.  If you or @h3xx can send me some test cases, I'll figure out where to put them.

**Comment by k-takata** [comment]
How about creating a copy of https://github.com/editorconfig/editorconfig-plugin-tests/tree/master/test_files with just adding a UTF-8 BOM to `test_files/.editorconfig`?

**Comment by h3xx** [comment]
+1 for this PR. Tested these changes against the issue in #194, and it in fact fixes the issue I described.

<details><summary>Summary of issue from #194 (~12 lines)</summary>

> For some reason, this plugin is causing `:e` to change the file's encoding option (`&fileencoding`) to utf-8, overriding whatever it's currently set to. This doesn't happen upon **first** opening the file, but **only** upon re-opening it with `:e`.
> 
> ## Steps to reproduce:
> 
> 1. Open and save a file in latin1 mode:
> 
> 
> ```shell
> export LANG=en_US LC_ALL=en_US LC_CTYPE=en_US
> vim foo.txt
> :set fenc=latin1
> :w
> ```
> 
> 2. Without closing Vim, reload the file:
> 
> 
> ```
> :e
> ```
> 
> 3. Check the file encoding hasn't changed:
> 
> 
> ```
> :echo &fenc
> ```
> 
> `&fenc` should still be `latin1`. If it was reset to `utf-8` the issue persists.

</details>

**Review by h3xx — APPROVED** [review]

**Comment by alkim0** [comment]
This will also fix https://github.com/editorconfig/editorconfig-vim/issues/196

**Comment by cxw42** [comment]
Merged --- let me know if you experience any issues!

**Pull request merged**